aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -0,0 +1,228 @@
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.glue.GlueClient;
+import software.amazon.awssdk.services.glue.model.ConcurrentModificationException;
+import software.amazon.awssdk.services.glue.model.CreateTableRequest;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetTableRequest;
+import software.amazon.awssdk.services.glue.model.GetTableResponse;
+import software.amazon.awssdk.services.glue.model.OperationTimeoutException;
+import software.amazon.awssdk.services.glue.model.ResourceNumberLimitExceededException;
+import software.amazon.awssdk.services.glue.model.Table;
+import software.amazon.awssdk.services.glue.model.TableInput;
+import software.amazon.awssdk.services.glue.model.UpdateTableRequest;
+
+public class GlueTableOperations extends BaseMetastoreTableOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(GlueTableOperations.class);
+
+  // same as org.apache.hadoop.hive.metastore.TableType.EXTERNAL_TABLE
+  // more details: https://docs.aws.amazon.com/glue/latest/webapi/API_TableInput.html
+  protected static final String GLUE_EXTERNAL_TABLE_TYPE = "EXTERNAL_TABLE";
+
+  private final Configuration conf;
+  private final GlueClient glue;
+  private final String catalogId;
+  private final boolean skipArchive;
+  private final String databaseName;
+  private final String tableName;
+  private final String fullName;
+
+  private FileIO fileIO;
+
+  /**
+   * All-arg constructor
+   * @param conf Hadoop config, passed in for users of HadoopIO
+   * @param glue Glue client
+   * @param catalogId Glue catalog ID, which is the AWS account ID. When null, it uses the account of the Glue client.
+   * @param skipArchive if Glue should skip an old table version when creating a new version in a commit.
+   *                    By default Glue archives all old table versions after an UpdateTable call.
+   *                    but Glue has a default max number of archived table versions (can be increased).
+   *                    So for streaming use case with lots of commits, it is recommended to turn this feature off.
+   * @param tableIdentifier table identifier
+   */
+  public GlueTableOperations(
+      Configuration conf,
+      GlueClient glue,
+      String catalogId,
+      boolean skipArchive,
+      TableIdentifier tableIdentifier) {
+    this.conf = conf;
+    this.glue = glue;
+    this.catalogId = catalogId;
+    this.skipArchive = skipArchive;
+    this.databaseName = IcebergToGlueConverter.getDatabaseName(tableIdentifier);
+    this.tableName = IcebergToGlueConverter.getTableName(tableIdentifier);
+    this.fullName = String.format("glue.%s.%s.%s", catalogId, databaseName, tableName);
```

Comment:
The full name here should be the name of the catalog, then the db and table name. This view is good for debugging (and may be what you want for `toString`) but the `name` returned for the table should be the name used to load it. So if I've called my Glue catalog `prodglue`, then it would be something like `prodglue.rdblue.test_table`.

Reply:
I see. I am basically using `glue.<catalogId>` as the name, that is why I am writing it like this. Sounds like the `name` concept is used here differently, let me add this additional field then.